### PR TITLE
README.md overhaul

### PR DESCRIPTION
**Pull request by bradenbest**
I overhauled the readme. Be sure to read the commit message.

Aside from re-working the sentences to be slightly less awkward, this fleshes out a lot of the details and adds more examples to further illustrate the points. I even added a couple section (`Preface` and `Troubleshooting`) to the installation instructions and re-worked all of that stuff to make it less confusing.

Ignore the first commit (`Phase 1 of 2: quick test`), and the commit messages for the latter two commits are the same.

If you have any comments, questions or concerns, please share.

### CODE REVIEW

**Comment by bradenbest** [comment]
I wanted to have it done in one commit, but  that sadly didn't work out quite as I had planned. So to make it easier to review, here are the differences:

[Main vs commit 1](https://github.com/ryukinix/mal/pull/97/commits/c8136b4808ed89a8887565c27a160a2890c3a021)

[commit 2 vs commit 3](https://github.com/ryukinix/mal/pull/97/commits/a5cad276b776aedb3f69e961c37e447c5e55d4f6)

The only differences between 1 and 2 are that I removed the "markdown testing" stuff and added in `$`'s to allow freedom to further flesh out comman-line examples (as expressed in the commit message).

Finally, [here's what it looks like when rendered](https://github.com/bradens-forks/mal/blob/master/README.md)

**Comment by bradenbest** [comment]
It occurs to me just now that I made a minor error in `Troubleshooting`:

    $ sudo ed $(which mal) <<< $'1s/python$/python3\nwq'
    28
    #!/usr/bin/python3
    29
    $ head -1 $(which mal)
    #!/usr/bin/python3

The `28/29` implies that the file is only 28 bytes long, this is because I used a test file to generate that output.

    #!/usr/bin/python
    ajhlsgldjg

Obviously not a big deal, but if ed's output bothers you at all, feel free to do this:

    $ sudo ed -s $(which mal) <<< $'1s/python$/python3\nwq'
             ^^^
    $ head -1 $(which mal)
    #!/usr/bin/python3

The `-s` puts ed in silent mode.

File: README.md [hunk 1]
```diff
@@ -46,42 +53,80 @@ More features are currently being developed! You can also request other features
...
+### Troubleshooting
+
+If you just *can't* get `mal` to run because it's crashing upon startup, make sure that everything is using `python3`
+
+    $ head -1 $(which mal)
+    #!/usr/bin/python
+    $ sudo ed $(which mal) <<< $'1s/python$/python3\nwq'
+    28
+    #!/usr/bin/python3
+    29
```
Comment (bradenbest) [inline README.md:110]: What I was referring to

**Review by ryukinix — APPROVED** [review]
This changes are ok. Sorry for the late review.

**Pull request merged**